### ISSUE TITLE
Enable more front-end user customization

### ISSUE THREAD

**Issue by sychang** [Recidiviz/pulse-dashboard#383]
Full context in [PRD](https://docs.google.com/document/d/1G2Ubey2siwkx1O_4p56jRXz1xOBvRu40EajdagnwPTk/edit#)

After launching the Revocations Matrix in MO and learning how they use the dashboard, there have been several requests for expansion of filters and additional filters. MO finds value in having more customization so they can drill down on specific slices of the population and identify anomalies and trends. In addition, the MVP PA feature to add an admission type filter ([PRD](https://docs.google.com/document/d/1314bPlAqvQGLFSUc9WNSZ7OUK52G3AoEKG6vPH3tgo8/edit#)) is dependent on this work.

There is currently a limit to how much slicing and customization a user can do on the dashboard before certain browsers (IE11) break or lose performance.

These are the set of known requests for data expansion and customization blocked on this that we have punted on for the time being. 

- P0: Adding Admission type filter to the top of the revocations matrix: Expand the dashboard beyond revocations to all admissions. 
- P1: Enable custom date range selection (#335)
Based on Fullstory and anecdotally, the time period filter is one of the most interacted with elements of this page. MO is keen to drill in on specific time periods to understand different trends over time.
- P1: Allow filtering by all 20+ case types in the MO dashboard, including things like electronic monitoring and deferred prosecution (today we show 4). 
- P2: Revocation cluster lookback period filter. Today, we set the default lookback period to 1 year. MO has requested the ability to filter by a shorter lookback period (1m, 3m, 6m, 9m). 
- P2: When time period = 1 month, show week over week data for the top chart (#339)
Related: When time period is 1 month, show the past 30 days vs. to the beginning of the month.

**Comment by sychang** [comment]
DONE!
